revscoring/extractors/api.py

```diff
@@ -13,19 +27,300 @@ def __init__(self, session, language=None):
+            if len(docs) > 0:
+                return docs[0]
+            else:
+                return None
+        else:
+            return None
+
+    def process_page_creation_doc(self, revision_metadata):
+        logger.info("Requesting page creation ({0}) from the API" \
+                    .format(revision_metadata.page_id))
+        docs = self.session.revisions.query(
+            pageids={revision_metadata.page_id},
+            direction="newer",
+            limit=1,
+            properties={'ids', 'user', 'timestamp', 'userid', 'comment',
+                        'flags', 'size'}
+        )
+        docs = list(docs)
+
+        if len(docs) == 1:
+            return docs[0]
+        else:
+            raise RevisionDocumentNotFound(
+                {'page_id': revision_metadata.page_id}
+            )
+
+    def process_user_doc(self, revision_metadata):
+        logger.info("Requesting user info ({0}) from the API" \
+                    .format(revision_metadata.user_text))
+        user_docs = self.session.users.query(
+            users={revision_metadata.user_text},
+            properties={'blockinfo', 'implicitgroups', 'groups', 'registration',
+                        'emailable', 'editcount', 'gender'}
+        )
+
+        user_docs = list(user_docs)
+
+        if len(user_docs) >= 1:
+            return user_docs[0]
+        else:
+            return None
+
+    @classmethod
+    def process_revision_metadata(cls, revision_doc):
+        return cls.revision_metadata_from_doc(revision_doc)
+
+    @classmethod
+    def process_user_info(cls, user_doc):
+        return cls.user_info_from_doc(user_doc)
+
+    @classmethod
+    def process_revision_text(cls, revision_doc):
+        if revision_doc is None: return None
+        return revision_doc.get('*', "")
+
+    @classmethod
+    def revision_metadata_from_doc(cls, rev_doc):
+        if rev_doc is None: return None
+        if 'timestamp' in rev_doc and rev_doc['timestamp'] is not None:
+            timestamp = Timestamp(rev_doc.get('timestamp'))
+        else:
+            timestamp = None
+
+        return RevisionMetadata(rev_doc.get('revid'),
+                                rev_doc.get('parentid'),
+                                rev_doc.get('user'),
+                                rev_doc.get('userid'),
+                                timestamp,
+                                rev_doc.get('comment'),
+                                rev_doc.get('page', {}).get('pageid'),
+                                rev_doc.get('page', {}).get('ns'),
+                                rev_doc.get('page', {}).get('title'),
+                                rev_doc.get('size'),
+                                'minor' in rev_doc)
+
+    @classmethod
+    def user_info_from_doc(cls, user_doc):
+        if user_doc is None: return None
+        try:
+            registration = Timestamp(user_doc.get('registration'))
+        except ValueError:
+            registration = None
```

Comment:
Is there any reason for using try/except here, but using if/else on line 267-270? Is any of the two methods preferred over the other?